src/text/text.rs

```diff
@@ -107,10 +107,25 @@ impl<S: Clone> Transform for Text<'_, S> {
 }
 
 impl<S: TextRenderer> Text<'_, S> {
+    fn line_height(&self) -> i32 {
+        self.text_style
+            .line_height
+            .to_absolute(self.character_style.line_height())
+            .saturating_as::<i32>()
+    }
+
     fn lines(&self) -> impl Iterator<Item = (&str, Point)> {
         let mut position = self.position;
 
-        self.text.lines().map(move |line| {
+        self.text.split_inclusive('\n').flat_map(move |mut line| {
+            let is_terminated = match line.strip_suffix('\n') {
+                None => false,
+                Some(s) => {
+                    line = s;
+                    true
+                }
+            };
```

Comment:
Simply using `self.text.split('\n')` should also work without the additional `is_terminated`and `extra_termination` code. The iterator will return an empty string for the trailing newline.

Reply:
> Thanks for the PR, this way of handling trailing newlines is better.
> 
> One issue with this change is that `lines()` did also support windows style line endings `\r\n`, which the implementation in this PR doesn't. But it wouldn't be hard to replicate this behavior: https://doc.rust-lang.org/src/core/str/mod.rs.html#2503-2505

Yeah, you could do the same `strip_suffix('\r')` => `line = s` as removing the terminator.

Reply:
> Simply using `self.text.split('\n')` should also work without the additional `is_terminated`and `extra_termination` code. The iterator will return an empty string for the trailing newline.

That sounds correct.  Originally I was attempting a different solution without using empty string iterations, so you had to do some math depending on the trailing terminator.

Reply:
> Yeah, you could do the same `strip_suffix('\r')` => `line = s` as removing the terminator.

You could, but the version from the core library will probably have slightly better performance, because it is byte based and not char based.